src/emails/templates/mjml.d.ts

```diff
@@ -2,8 +2,72 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-declare module "*.mjml?raw" {
```

Comment:
Would be nice if we didn't have to maintain this... there is https://www.npmjs.com/package/@types/mjml but that's for the (server-side) function definitions. I guess from https://github.com/mjmlio/mjml/issues/930 there is probably not much interest in them maintaining this upstream, but we'll need to make sure to keep it up to date - the TS compiling will not guarantee that the template will be rendered to HTML.

Reply:
Yeah, I was also looking at that. There is also [mjml-react](https://www.npmjs.com/package/mjml-react) for which (2-years-old) type definitions are available, but that does the whole thing where it creates specialised component wrappers that convert camelCased attributes to kebab-cased ones, which makes it more of a blocking dependency when it's not updated.

So I don't have a _strong_ opinion, but I would be leaning towards just adding our own type definitions. I think I can cover basically the entire API in half an hour (I just didn't do it yet since I didn't know we'd use this), and we'd only really have to update it if we want to use a changed API in the future - which would be a matter of seconds to do, and we'd be in control ourselves.

Reply:
Took that half-hour just now to finish these up: 1e8ae4eaf

I think that should allow us to create all the templates we want for the foreseeable future.